back/app/services/survey_results_generator_service.rb

```diff
@@ -13,30 +15,18 @@ def generate_submission_count
     { totalSubmissions: inputs.size }
   end
 
-  def generate_results(field_id: nil, group_mode: nil, group_field_id: nil)
-    if !field_id
-      # Return all results (ungrouped)
-      results = fields.filter_map do |field|
-        visit field
+  def generate_results(field_id: nil)
+    if field_id
+      field = find_question(field_id)
+      visit field
+    else
+      results = fields.filter_map do |field_|
+        visit field_
```

Comment:
why is this `field_` and not `field`?

Reply:
Rubocop says

> Shadowing outer local variable - `field`.

I'll probably better use `f` to avoid such confusion.